Branch-SDK/src/io/branch/referral/PrefHelper.java

```diff
@@ -332,10 +333,22 @@ public String getAppKey() {
     }
 
 	public void setBranchKey(String key) {
-	    Branch_Key = key;
-    }
+		Branch_Key = key;
+		String currentBranchKey = getString(KEY_BRANCH_KEY);
+		if (key == null || currentBranchKey == null || !currentBranchKey.equals(key)) {
```

Comment:
This should be checking against the *previous* Branch key as well. The key isn't likely going to change during the actual app life time, it will more likely be changing between runs. Note in the iOS SDK, it keeps track of the last used Branch key, and compares against that.